### ISSUE TITLE
to_netcdf() should not implicitly load dask arrays of strings into memory

### ISSUE THREAD

**Issue by shoyer** [pydata/xarray#2069]
As discussed in https://github.com/pydata/xarray/pull/2058#discussion_r181606513, we should have an explicit interface of some sort, either via encoding or some new keyword argument to `to_netcdf()`.

**Comment by max-sixty** [comment]
Trying to keep us below 1K issues — is this still current?

**Comment by shoyer** [comment]
Yes, this issue refers to the "TODO" note on these lines of code, which is triggered when reading a fixed length string data into a netCDF files:
https://github.com/pydata/xarray/blob/56209bd9a3192e4f1e82c21e5ffcf4c3bacaaae3/xarray/coding/strings.py#L73-L74

`encode_string_array()` still eagerly loads data into memory.

That said -- this is only the default for netCDF3 files. Most people writing large files and concerned about streaming writes are probably using Zarr these days. So in practice, I'm not sure anyone cares enough to fix this issue.
